### PR TITLE
Backport changes to track 2.0 to fix CI

### PR DESCRIPTION
**Pull request by DnPlas**
This PR includes the following changes and aims at backporting them to track/2.0:

* ci: update workflows directory
* fix: Pin COS charms in integration tests to latest/stable (#108)
* build: bump jinja2 3.1.x -> 3.1.4 (#118)
* build, ci: bump juju 2.9 -> 3.x
* ci, build: bump python-libjuju 3.2 -> 3.4

### CODE REVIEW

File: requirements-unit.txt [hunk 1]
```diff
@@ -53,13 +53,11 @@ idna==3.4
...
-kubernetes==26.1.0
-    # via -r requirements.txt
-lightkube==0.12.0
-    # via -r requirements.txt
-lightkube-models==1.26.0.4
+jinja2==3.1.4
+    # via -r requirements.in
+kubernetes==27.2.0
+    # via -r requirements.in
+lightkube==0.14.0
```
Comment (orfeas-k) [inline requirements-unit.txt:60]: Do we need this?

Comment (DnPlas) [reply requirements-unit.txt:60]: No, will update.

**Review by rgildein — DISMISSED** [review]
LGTM, only one comment.

**Review by rgildein — APPROVED** [review]
LGTM, thanks!

**Pull request merged**